### PR TITLE
Adding ElasticIP resource

### PR DESCRIPTION
**Pull request by enderv**
Signed-off-by: chris <chris.vodak@gmail.com>

### Description of your changes

Fixes https://github.com/crossplane/provider-aws/issues/260 adding support for ElasticIP
### Checklist

I have:
- [x] Run `make reviewable` to ensure this PR is ready for review.
- [x] Ensured this PR contains a neat, self documenting set of commits.
- [ ] Updated any relevant [documentation], [examples], or [release notes].
- [ ] Updated the dependencies in [`app.yaml`] to include any new role permissions.

[documentation]: https://github.com/crossplane/crossplane/tree/master/docs
[examples]: https://github.com/crossplane/crossplane/tree/master/cluster/examples
[release notes]: https://github.com/crossplane/crossplane/tree/master/PendingReleaseNotes.md
[`app.yaml`]: https://github.com/crossplane/provider-aws/blob/master/config/package/manifests/app.yaml

### CODE REVIEW

**Comment by enderv** [comment]
Works for me. Working on it now

File: apis/ec2/v1alpha1/elasticip_types.go [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+	// +optional
+	// +immutable
+	CustomerOwnedIPv4Pool *string `json:"customerOwnedIPv4Pool,omitempty"`
+
+	// Set to vpc to allocate the address for use with instances in a VPC.
+	// Default: The address is for use with instances in EC2-Classic.
+	// +optional
+	// +immutable
+	// +kubebuilder:validation:Enum=vpc;standard
+	Domain string `json:"domain,omitempty"`
```
[the remark this answers is not in the document]
Comment (enderv) [reply apis/ec2/v1alpha1/elasticip_types.go]: Oops yeah changed it over to `*string`

File: pkg/controller/ec2/elasticip/controller.go [hunk 2]
```diff
@@ -0,0 +1,273 @@
...
+	current := cr.Spec.ForProvider.DeepCopy()
+	ec2.LateInitializeElasticIP(&cr.Spec.ForProvider, &observed)
+	if !cmp.Equal(current, &cr.Spec.ForProvider) {
+		if err := e.kube.Update(ctx, cr); err != nil {
+			return managed.ExternalObservation{}, errors.Wrap(err, errSpecUpdate)
+		}
+	}
+
+	// TODO(cvodak) ? this was copied
+	cr.SetConditions(runtimev1alpha1.Available())
```
Comment (hasheddan) [inline pkg/controller/ec2/elasticip/controller.go:167]: Feel free to remove this `TODO` :)

Comment (enderv) [reply pkg/controller/ec2/elasticip/controller.go]: oof. done

**Pull request merged**